### ISSUE TITLE
Rustdoc doesn't have "Skip Navigation" links

### ISSUE THREAD

**Issue by JayanAXHF** [rust-lang/rust#151420]
# Problem Synopsis
Skip navigation links allow users using keyboard navigation (i.e. the `Tab` key) to be able to skip long repeated sections and get directly to the main content. Currently, while using `Tab` navigation, one has to get past the entire left contents bar, before being able to access the main content (see video demo below). WCAG Success Criterion 2.4.1 (Level A) states 
> A [mechanism](https://www.w3.org/TR/WCAG22/#dfn-mechanism) is available to bypass blocks of content that are repeated on multiple [web pages](https://www.w3.org/TR/WCAG22/#dfn-web-page-s).

This feature is currently missing within `rustdoc`, making keyboard navigation a pain to work with on large crates.

## Useful links
1. https://webaim.org/techniques/skipnav/#other
2. https://webaim.org/standards/wcag/checklist#sc2.4.1
3. https://www.w3.org/TR/WCAG22/#bypass-blocks

# Additional Details

https://github.com/user-attachments/assets/0c454729-2602-43a1-9b3b-ffc374178371

**Comment by ThanhNguyxn** [comment]
@rustbot claim

I'm working on implementing a "Skip to main content" link for this issue. I'll start with a basic implementation that allows users to skip directly to the `#main-content` section, and we can iterate on adding more skip links (e.g., between the page summary and module contents) based on feedback.
